server.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+require('dotenv').config();
+const express = require('express');
+const cors = require('cors');
+const weatherListing = require('./data/weather.json');
+
+const app = express(); 
+
+app.use(cors());
+
+const PORT = process.env.PORT || 3001
+
+app.get('/hello', (request, response) => { response.send('Hello, it works!') })
+app.get('/data/weather', handleGetWeatherList);
+
+function handleGetWeatherList(req, res) {
+    req.query
+    console.log('The shopping list route was hit!!!');
+    res.status(200).send(weatherListing)
+}
```

Comment:
Hi there Kirk, nice job on getting server listening :)

For this you are on the right track here. You have to use `req.query.searchQuery`. You may find that you need to assign it to a variable.

From there, follow task cards. Use the .find() method to discover which city the lat, lon and searchQuery information belong to.

Then you will do your conditionals, so if (!city) do something (send error) else, define the new weather array and map over your data (this is where your constructor Function `Forecast` will come into play (or ES6 class `Forecast`) 👍